lib/archethic/account/mem_tables/token_ledger.ex

```diff
@@ -131,49 +218,24 @@ defmodule Archethic.Account.MemTables.TokenLedger do
     end)
   end
 
-  @doc """
-  Spend all the unspent outputs for the given address
-  """
-  @spec spend_all_unspent_outputs(binary()) :: :ok
-  def spend_all_unspent_outputs(address) do
-    case :ets.lookup(@unspent_output_index_table, address) do
-      [] ->
-        :ok
-
-      utxos ->
-        {:ok, pid} = DB.start_inputs_writer(:token, address)
-
-        Enum.each(utxos, fn {to, from, token_address, token_id} ->
-          [{_, amount, _, timestamp, protocol_version}] =
-            :ets.lookup(@ledger_table, {to, from, token_address, token_id})
-
-          DB.append_input(pid, %VersionedTransactionInput{
-            protocol_version: protocol_version,
-            input: %TransactionInput{
-              from: from,
-              amount: amount,
-              spent?: true,
-              timestamp: timestamp,
-              type: {:token, token_address, token_id}
-            }
-          })
-
-          :ets.delete(@ledger_table, {address, from, token_address, token_id})
-        end)
-
-        :ets.delete(@unspent_output_index_table, address)
-
-        DB.stop_inputs_writer(pid)
-    end
-  end
-
   @doc """
   Retrieve the entire inputs for a given address (spent or unspent)
   """
   @spec get_inputs(binary()) :: list(VersionedTransactionInput.t())
   def get_inputs(address) when is_binary(address) do
     case :ets.lookup(@unspent_output_index_table, address) do
       [] ->
+        :token
+        |> DB.get_inputs(address)
+
+      [{^address, true}] ->
+        :token
+        |> DB.get_inputs(address)
+        |> Enum.map(
+          &%VersionedTransactionInput{&1 | input: %TransactionInput{&1.input | spent?: false}}
+        )
+
+      [{^address, true}] ->
```

Comment:
I don't understand the need of this case statement as the previous one is already covering the case.